homeassistant/components/light/ads.py

```diff
@@ -0,0 +1,131 @@
+    ads_var_brightness = config.get(CONF_ADS_VAR_BRIGHTNESS)
+    name = config.get(CONF_NAME)
+
+    add_devices([AdsLight(ads_hub, ads_var_enable, ads_var_brightness,
+                          name)], True)
+
+
+class AdsLight(Light):
+    """Representation of ADS light."""
+
+    def __init__(self, ads_hub, ads_var_enable, ads_var_brightness, name):
+        """Initialize AdsLight entity."""
+        self._ads_hub = ads_hub
+        self._on_state = False
+        self._brightness = 50
+        self._name = name
+        self.ads_var_enable = ads_var_enable
+        self.ads_var_brightness = ads_var_brightness
+        self.stype = 'dimmer'
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register device notification."""
+        @callback
+        def async_update_on_state(name, value):
+            """Handle device notifications for state."""
+            _LOGGER.debug('Variable %s changed its value to %d', name, value)
+            self._on_state = value
+            self.async_schedule_update_ha_state()
+
+        @callback
+        def async_update_brightness(name, value):
+            """Handle device notification for brightness."""
+            _LOGGER.debug('Variable %s changed its value to %d', name, value)
+            self._brightness = value
+            self.async_schedule_update_ha_state()
+
+        self._ads_hub.add_device_notification(
+            self.ads_var_enable, self._ads_hub.PLCTYPE_BOOL,
+            async_update_on_state
+        )
+        self._ads_hub.add_device_notification(
+            self.ads_var_brightness, self._ads_hub.PLCTYPE_INT,
+            async_update_brightness
+        )
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def brightness(self):
+        """Brightness of the light."""
+        return self._brightness
+
+    @property
+    def is_on(self):
+        """If light is on."""
+        return self._on_state
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_ADS
+
+    def turn_on(self, **kwargs):
+        """Turn the light on or set a specific dimmer value."""
+        brightness = kwargs.get(ATTR_BRIGHTNESS)
+        if brightness is not None:
+            self._brightness = brightness
+
+        self._ads_hub.write_by_name(self.ads_var_enable, True,
+                                    self._ads_hub.PLCTYPE_BOOL)
+
+        if self.ads_var_brightness is not None:
+            self._ads_hub.write_by_name(
+                self.ads_var_brightness, self._brightness,
+                self._ads_hub.PLCTYPE_UINT
+            )
+
+        self._on_state = True
+
+    def turn_off(self, **kwargs):
+        """Turn the light off."""
+        self._ads_hub.write_by_name(self.ads_var_enable, False,
+                                    self._ads_hub.PLCTYPE_BOOL)
+        self._on_state = False
+
+    def update(self):
```

Comment:
You can remove this since you use the callbacks instead.

Reply:
Done.